app/assets/javascripts/courseEdit/components/ComponentEdit.js

```diff
@@ -171,13 +171,15 @@ class ComponentEdit extends React.Component {
           subsectionId={this.subsectionId}
           component={this.state.component}
           callback={this.onFormCompletion}
+          disabled={false}
         />
 
         <DeleteModal
           openDeleteModal={this.state.openDeleteModal}
           closeModal={this.closeDeleteModal}
           deleteFunction={this.deleteComponent}
           objectType='component'
+          disabled={this.props.course.isPublished}
```

Comment:
In the future if a property of a prop object is used and is pretty important for the page's functionality you should add an appropriate PropTypes validation to be more thorough. For example, if I have `this.props.course` and I must have `this.props.course.isPublished` to ensure functionality then my PropTypes check for course would look like:

```javascript
ComponentEdit.propTypes = {
  course: PropTypes.shape({
    disabled: PropTypes.bool.isRequired, 
  }), 
}
```
And there is a similar thing to do for objects in arrays.